src/cmd/services/m3coordinator/handler/placement/get.go

```diff
@@ -40,20 +41,23 @@ const (
 type getHandler Handler
 
 // NewGetHandler returns a new instance of a placement get handler.
-func NewGetHandler(service placement.Service) http.Handler {
-	return &getHandler{service: service}
+func NewGetHandler(client clusterclient.Client, cfg config.Configuration) http.Handler {
+	return &getHandler{client: client, cfg: cfg}
 }
 
 func (h *getHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	logger := logging.WithContext(ctx)
 
-	placement, version, err := h.service.Placement()
+	service, err := Service(h.client, h.cfg)
+	if err != nil {
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	placement, version, err := service.Placement()
 	if err != nil {
-		// An error from `get` signifies "key not found", meaning there is
-		// no placement and as such, should not be treated as an actual error to
-		// the user.
-		w.Write([]byte("no placement found\n"))
+		handler.Error(w, errors.New("placement not found"), http.StatusBadRequest)
```

Comment:
Is trying to get the placement when there isn't one actually a `400 Bad Request`? Maybe, just a thought.

Reply:
Oh true, yeah makes sense.  I'll restore this to return 200.